fluss-client/src/main/java/com/alibaba/fluss/client/write/Sender.java

```diff
@@ -216,7 +225,9 @@ private void sendWriteData() throws Exception {
         if (!batches.isEmpty()) {
             addToInflightBatches(batches);
 
-            // TODO add logic for batch expire.
+            // check and expire batches if they have reached batch delivery timeout. This can avoid
+            // the client to wait for a long time while the batch is forever retry.
+            checkAndExpireBatches(now);
```

Comment:
What happens if the batch is expired here but is still in the `sendWriteRequests(batches)`?  Will the expired batch still be sent to servers? What happens if the batches fail or success from the request?